### PR TITLE
Documentation work, removed unneeded models

### PR DESCRIPTION
**Pull request by TristanAlbers**
- Removed unneeded models
- Added info to dev guide
- Documented typesystem and constraint

### CODE REVIEW

**Comment by TristanAlbers** [comment]
When remaking models (if needed) be aware of deprecated links, Flintmodel UML not complete

File: docs/TYPESYSTEM_CONSTRAINS.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+Acts/Facts/Duties must have unique names. If this is not the case an error will be generated.
+
+### ActiveActorsRule (Error)
+The maximum active actors inside the Flintrunner is declared as 2. Adding more will generate this error.
+
+### ArchivedUseRule (Error)
+When Acts/Facts/Duties are archived they cannot be used inside an active role in the Flintmodel, meaning no references should be made to them. If they are used in the active model an error wil be generated.
+
+### ObligatoryRoleEmptyAct (Error)
+Acts have certain properties which must be filled. The properties are `actor`,`object` and `recipient`. When one of these is not filled in an error will be generated.
```
[the remark this answers is not in the document]
Comment (TristanAlbers) [reply docs/TYPESYSTEM_CONSTRAINS.md:50]: Documentation about implemenation put inside codebase [in this pullrequest](https://github.com/discipl/flinteditor-mps/pull/202)

File: docs/uml/flintmodel-model.puml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+@startuml FlintModel-Model
+skinparam nodesep 50
+skinparam ranksep 150
+
+interface "Flint.structure.IFlintModel" <INamedConcept>
```
Comment (blackstardlb) [inline docs/uml/flintmodel-model.puml:5]: include these methods in IFlintModel

image

File: docs/uml/ihassources-model.puml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+"Flint.structure.IExplainable" : explanation : string
+interface "FlintCellModels.structure.IProvidesRoleTagsCell" 
+interface "Flint.structure.IHasLanguage" <INamedConcept>
+interface "Flint.structure.IHasSources" 
+interface "de.slisson.mps.richtext.structure.IWord"
+class "Flint.structure.Fact"  << (R, lightgrey) >>
+class "Flint.structure.Act"  << (R, lightgrey) >>
+class "Flint.structure.Duty"  << (R, lightgrey) >> 
+class "Flint.structure.FlintSource"  
+"Flint.structure.FlintSource" : text : string
```
Comment (blackstardlb) [inline docs/uml/ihassources-model.puml:17]: text is deprecated

**Review by blackstardlb — CHANGES_REQUESTED** [review]
Add Titles to each diagram 

```
title Act Model
````

**Comment by blackstardlb** [comment]
I've added installLocalGitHook task to gradle which after run once will install a precommit hook that auto generates the uml diagram renders

**Review by blackstardlb — APPROVED** [review]
lgtm 👍

**Pull request merged**